backend-shared/plannershared.cpp

```diff
@@ -113,3 +113,13 @@ void plannerShared::set_vpmb_conservatism(int value)
 {
 	DivePlannerPointsModel::instance()->setVpmbConservatism(value);
 }
+
+bool plannerShared::dobailout()
+{
+	return qPrefDivePlanner::dobailout();
+}
+void plannerShared::set_dobailout(bool value)
+{
+	qPrefDivePlanner::set_dobailout(value);
+	DivePlannerPointsModel::instance()->emitDataChanged();
```

Comment:
so why is this done differently from the cases in the previous commit, where you have this asymmetric pattern of reading from qPref and calling the setter in `DivePlannerPointsModel` ?
Calling that signal in the unrelated class seems odd. But maybe there's a good reason to do it this way that I just don't see?

Reply:
I too find it odd, but desktop-widgets/diveplanner.cpp have a number of case like this:

void PlannerSettingsWidget::setBackgasBreaks(bool dobreaks)
 {
-       qPrefDivePlanner::instance()->set_doo2breaks(dobreaks);
-       plannerModel->emitDataChanged();
+       plannerShared::set_doo2breaks(dobreaks);
 }

Where you see I move the functionality 1-1 to backend-shared.

Actually I do not know if there is a good reason.

Reply:
That's a good explanation. @atdotde @bstoeger - any insights into why we do things that way?
(I think Berthold is offline, now that I wrote this)